tests/main/apparmor-prompting-integration-tests/task.yaml

```diff
@@ -0,0 +1,128 @@
+    the prompting-client snap in scripted mode, check that those prompts match
+    what is expected, and that the client carries out the operations as intended.
+
+    Details about the operation of the prompting client in scripted mode:
+    https://github.com/canonical/prompting-client/blob/main/docs/running-the-scripted-client.md
+
+
+systems:
+  - ubuntu-16*
+  - ubuntu-18*
+  - ubuntu-2*
+  - ubuntu-core-*
+
+environment:
+    VARIANT/read_single_allow: read_single_allow
+    VARIANT/read_single_deny: read_single_deny
+    VARIANT/write_single_allow: write_single_allow
+    VARIANT/write_single_deny: write_single_deny
+    VARIANT/create_multiple_allow: create_multiple_allow
+    VARIANT/create_multiple_deny: create_multiple_deny
+    VARIANT/download_file_defaults: download_file_defaults
+    VARIANT/download_file_safer: download_file_safer
+    VARIANT/download_file_conflict: download_file_conflict
+    VARIANT/timespan_allow: timespan_allow
+    VARIANT/timespan_deny: timespan_deny
+    VARIANT/create_multiple_actioned_by_other_pid_always_allow: create_multiple_actioned_by_other_pid_always_allow
+    VARIANT/create_multiple_actioned_by_other_pid_always_deny: create_multiple_actioned_by_other_pid_always_deny
+    VARIANT/create_multiple_not_actioned_by_other_pid_single_allow: create_multiple_not_actioned_by_other_pid_single_allow
+    VARIANT/create_multiple_not_actioned_by_other_pid_single_deny: create_multiple_not_actioned_by_other_pid_single_deny
+    VARIANT/write_read_multiple_actioned_by_other_pid_allow_deny: write_read_multiple_actioned_by_other_pid_allow_deny
+    VARIANT/write_read_multiple_actioned_by_other_pid_deny_allow: write_read_multiple_actioned_by_other_pid_deny_allow
+
+    TIMEOUT: "30" # Define common timeout which can be modified as needed
+
+prepare: |
+    tests.session prepare -u test
+    tests.session -u test exec sh -c 'mkdir -p "/home/test/integration-tests"'
+    snap install prompting-client
+
+restore: |
+    snap set system experimental.apparmor-prompting=false
+    tests.session -u test exec sh -c 'rm -rf "/home/test/integration-tests"'
+    tests.session restore -u test
+
+debug: |
+    TEST_UID="$(id -u test)"
+    echo "outstanding prompts:"
+    snap debug api "/v2/interfaces/requests/prompts?user-id=$TEST_UID" || true
+    echo "rules:"
+    snap debug api "/v2/interfaces/requests/rules?user-id=$TEST_UID" || true
+
+execute: |
+    echo "Precondition check that snapd is active"
+    systemctl is-active snapd.service snapd.socket
+    SNAPD_PID="$(systemctl show --property MainPID snapd.service | cut -f2 -d=)"
+
+    echo "Enable prompting via snap client where possible"
+    # Prompting is unsupported everywhere but the Ubuntu non-core systems with
+    # kernels which support apparmor prompting
+    if ! os.query is-ubuntu || os.query is-core || NOMATCH 'prompt' < /sys/kernel/security/apparmor/features/policy/permstable32 ; then
```

Comment:
`NOMATCH 'prompt' < /sys/kernel/security/apparmor/features/policy/permstable32` is not considered when `/sys/kernel/security/apparmor/features/policy/permstable32` doesn't exist.

Check: https://github.com/canonical/snapd/actions/runs/11386138513/job/31678432632?pr=14518#step:15:4157

Reply:
Good catch, thank you! I'll add this here and make sure the existing spread tests around prompting handle this correctly too.

Reply:
It seems `! grep 'prompt' /sys/kernel/security/apparmor/features/policy/permstable32` handles this properly even if the file doesn't exist, so I'm inclined to return to that instead. Or at least leave them be in the existing spread tests. I can do `! [ -f /sys/kerne/security/apparmor/features/policy/permstable32 ] || NOMATCH 'prompt' < /sys/kerne/security/apparmor/features/policy/permstable32`.